File: OddOccurrencesInArray.js
// OddOccurrencesInArray solution
function solution(A) {
    let unpaired = {};
    for (let i = 0; i < A.length; i++) {
        if (unpaired[A[i]]) {
            unpaired[A[i]] = false;
        } else {
            unpaired[A[i]] = true;
        }
        console.log(unpaired);
    }
    for (var i in unpaired) {
        if (unpaired[i]) {
           return parseInt(i);;
        }
    }
}

File: cpp/OddOccurrencesInArray.cpp
// you can use includes, for example:
// #include <algorithm>
#include <unordered_map>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    std::unordered_map<int, bool> unpaired;
    auto A_it = A.begin();
    while (A_it != A.end()) {
        if (!unpaired[*A_it]) {
            unpaired[*A_it] = true;
        } else {
            unpaired[*A_it] = false;
        }
        A_it++;
    }
    for (auto const& pair: unpaired) {
        if (pair.second)
            return pair.first;
    }
    return 0;
}
